tests/server/controllers/userController.test.js

```diff
@@ -210,4 +280,329 @@ describe("User Controller", () => {
                 })
         })
     })
+
+    describe('Update Users', () => {
+      it('should not allow access if not logged on', (done) => {
+        chai.request(app)
+            .post(updateUserUrl)
+            .end((err, res) => {
+              res.status.should.be.eql(401);
+              res.body.error.should.be.eql('Please login');
+              done();
+            })
+      })
+      it('should not allow students to update users', (done) => {
+        let cookie;
+        chai.request(app)
+            .post(loginUrl)
+            .send({ email: 'student@gmail.com', password: '1234567' })
+            .then((res) => {
+                userSession = res.body.userSession
+                cookie = mockSession('session', process.env.SECRET, userSession);
+                res.body.should.be.eql({
+                    status: "success",
```

Comment:
Same thing here. Each `it` hook looks complicated to me. I notice you have two requests under one `it` hook. I think you give each request an `it` hooks

Reply:
Hey I'm not sure we need a `it('should login successfully')` hook each time, as the login is not the thing we're testing for (we have a test for it earlier, and we also make sure to check that each subsequent login is successful with `res.body.should.be.eql({})`)

This is using a similar structure in the `chai-http` docs:

```
var agent = chai.request.agent(app)
agent
  .post('/session')
  .send({ username: 'me', password: '123' })
  .then(function (res) {
    expect(res).to.have.cookie('sessionid');
    // The `agent` now has the sessionid cookie saved, and will send it
    // back to the server in the next request:
    return agent.get('/user/me')
      .then(function (res) {
         expect(res).to.have.status(200);
      });
  });
```

Reply:
Cool. Can you add some comments on the tests just like as we have it in the sample test shown above? @jamsgra